# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/internal_types.rs
# Commit: 0c03e3ad5e56
# Full Hash: 0c03e3ad5e56201e35437734b0ae1d34faa52005
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-15 09:29:54
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 1 - Remove PrevPassAlpha and PrevPassColor r=nical
#   
#   Remove usage of the implicit prev pass alpha and color texture
#   samplers from batching / renderer / shader code. They are replaced
#   by explicit references to the texture ID for the source task.
# ==============================================================================

diff -r 4e0ba157c016 -r 0c03e3ad5e56 gfx/wr/webrender/src/internal_types.rs
--- a/gfx/wr/webrender/src/internal_types.rs	Sat Dec 12 22:13:43 2020 +0000
+++ b/gfx/wr/webrender/src/internal_types.rs	Mon Dec 14 19:38:48 2020 +0000
@@ -268,10 +268,6 @@
     TextureCache(CacheTextureId, ImageBufferKind, Swizzle),
     /// An external image texture, mananged by the embedding.
     External(DeferredResolveIndex, ImageBufferKind),
-    /// The alpha target of the immediately-preceding pass.
-    PrevPassAlpha,
-    /// The color target of the immediately-preceding pass.
-    PrevPassColor,
     /// Select a dummy 1x1 white texture. This can be used by image
     /// shaders that want to draw a solid color.
     Dummy,
@@ -285,10 +281,7 @@
             TextureSource::External(_, image_buffer_kind) => image_buffer_kind,
 
             // Render tasks use texture arrays for now.
-            TextureSource::PrevPassAlpha
-            | TextureSource::PrevPassColor
-            | TextureSource::Dummy => ImageBufferKind::Texture2DArray,
-
+            TextureSource::Dummy => ImageBufferKind::Texture2DArray,
 
             TextureSource::Invalid => ImageBufferKind::Texture2D,
         }